app/controllers/dashboards_controller.rb

```diff
@@ -22,9 +31,28 @@ def export
     redirect_to dashboards_path, alert: e.message
   end
 
+  def push
+    version = Version.preview_version
+
+    if version.blank?
+      flash.alert = 'No preview version available'
+    else
+      pv = Version.create(number: version.number, production: true, user: current_user)
+
+      if pv.persisted?
+        flash.notice = 'Production data updated'
+      else
+        flash.alert = 'Production data not updated, remains at previous production version'
+      end
+    end
+
+    redirect_to dashboards_path
```

Comment:
Similar comment here. Since push is a non-idempotent action, this should be a POST or PUT request. It is also unclear to me why we would redirect to dashboards path and essentially clear the params provided when we have an "error" -- I would think in your else clause when the error condition results you would do a render instead of redirect.